kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/package-info.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+/**
+ *
+ * <h2 id='implementing'>Implementing Filters</h2>
+ *
+ * <h3 id='implementing.filterResults'>Filter Results</h3>
+ * <p>Filter implementation must return a {@link java.util.concurrent.CompletionStage} containing a
+ * {@link io.kroxylicious.proxy.filter.FilterResult} object. It is the job of FilterResult to convey what
+ * message is to forwarded to the next filter in the chain (or broker /client if at the chain's beginning
```

Comment:
`forwarded` -> `be forwarded`

I feel like we should lead with the idea that the FilterResult is an instruction to the framework, as you said below. The FilterResult instructs the framework what to do after the user has finished doing their work in the protocol filter. Options including: forward a message, close connection, drop message. Though I get that we should emphasise the message forwarding part. So I'd rephrase it something like:

`It is the job of FilterResult to carry an instruction to the framework, describing what should be done when the custom protocol filter has finished operating on the message. The most common instruction is to forward the message to the next filter in the chain. Other options include closing the connection or dropping the message.`